api/organisations/chargebee/chargebee.py

```diff
@@ -88,3 +88,20 @@ def get_hosted_page_url_for_subscription_upgrade(
     params = {"subscription": {"id": subscription_id, "plan_id": plan_id}}
     checkout_existing_response = chargebee.HostedPage.checkout_existing(params)
     return checkout_existing_response.hosted_page.url
+
+
+def get_subscription_metadata(subscription_id: str) -> ChargebeeObjMetadata:
+    subscription = chargebee.Subscription.retrieve(subscription_id).subscription
+    addon_ids = (
+        [addon.id for addon in subscription.addons] if subscription.addons else []
+    )
+
+    chargebee_cache = ChargebeeCache()
+    plan_metadata = chargebee_cache.plans[subscription.plan_id]
```

Comment:
This should return a ChargebeeMetaData object, right? Could we add a method to the ChargebeeCache class which does something like: 

```
def get_plan_metadata(plan_id: str) -> ChargebeeMetaData:
    return self.plans[plan_id]
```

Just so that we have the type hint really. 

Also, what if the plan id doesn't exist? Should we handle that?

Reply:
> Also, what if the plan id doesn't exist? Should we handle that?

I don't think we should, it should fail, and it should fail loudly

Reply:
> This should return a ChargebeeMetaData object, right? Could we add a method to the ChargebeeCache class which does something like:

Well, that's how I did it but, that problem with that is I can't do that with addons(i.e: I can't create a method called `get_addon_metadata` because we are going to call the multiple times and every call is a database call) and it makes the interface look sort of strange/uneven. Does that make sense?

Reply:
I do agree... but should it fail with a more descriptive exception? Maybe if we create a custom exception (but don't subclass from APIException). Perhaps it's overkill - I guess as long as we have sentry, it should be fine as we'll get the full traceback.